Update security policy regarding backporting patches

cary-ilm opened the pull request:
This wording captures our policy: we commit to fixes for main and the most recent minor release, but prior to that, discussed on a case-by-case basis. Better to leave it at than than attempt to be more specific, since those specifics can be determined based on the situation.

cary-ilm committed:
Update security policy regarding backporting patches

This wording captures our policy: we commit to fixes for main and the
most recent minor release, but prior to that, discussed on a
case-by-case basis. Better to leave it at than than attempt to be more
specific, since those specifics can be determined based on the situation.

Signed-off-by: Cary Phillips <cary@ilm.com>

lgritz commented on SECURITY.md:
```diff
@@ -61,17 +61,18 @@ See the [release notes](CHANGES.md) for more information.
-| 3.0.x    | :warning: Only the most critical fixes, only if they can be easily backported. |
-| 2.5.x    | :warning: Only the most critical fixes, only if they can be easily backported. |
-| <= 1.x   | :x: No longer receiving patches of any kind. |
+| main      | :white_check_mark: :construction: All fixes immediately, although this branch is under development with potential unstabe ABI/API
```
```suggestion
| main      | :white_check_mark: :construction: All fixes immediately, although this branch is under development with potential unstable ABI/API
```

lgritz commented on SECURITY.md:
```diff
@@ -61,17 +61,18 @@ See the [release notes](CHANGES.md) for more information.
+| 3.1.x    | :warning: Patch releases considered in response to specific requests
+| 3.0.x    | :warning: Patch releases considered in response to specific requests
+| 2.5.x    | :warning: Patch releases considered in response to specific requests
+| <= 1.x   | :x: No longer receiving patches of any kind |
```
What about 2.0-2.4?

cary-ilm committed:
Support policy for 2.0-2.4:

2.4 was the first release by the ASWF, recent enough to negotiate
patches. Prior to that our knowledge is so limited, and CMake support
was so fragile, that we should not even entertain the possibility.

Signed-off-by: Cary Phillips <cary@ilm.com>
